ui/src/hooks/useKeyboard.ts

```diff
@@ -111,12 +114,60 @@ export default function useKeyboard() {
     }
   };
 
+  const KEEPALIVE_INTERVAL = 50;
+
+  const cancelKeepAlive = useCallback(() => {
+    if (keepAliveTimerRef.current) {
+      clearInterval(keepAliveTimerRef.current);
+      keepAliveTimerRef.current = null;
+    }
+  }, []);
+
+  const scheduleKeepAlive = useCallback(() => {
+    // Clear existing timer if it exists
+    if (keepAliveTimerRef.current) {
+      clearInterval(keepAliveTimerRef.current);
```

Comment:
should we also `delete keepAliveTimerRef.current;` to ensure there's not a reference cycle?

Reply:
Calling `cancelKeepAlive()` instead.